pkg/store/gcworker/gc_worker.go

```diff
@@ -805,73 +806,93 @@ func (w *GCWorker) runGCJob(ctx context.Context, safePoint uint64, concurrency i
 func (w *GCWorker) deleteRanges(ctx context.Context, safePoint uint64, concurrency int) error {
 	metrics.GCWorkerCounter.WithLabelValues("delete_range").Inc()
 
-	se := createSession(w.store)
-	defer se.Close()
-	ranges, err := util.LoadDeleteRanges(ctx, se, safePoint)
+	s := createSession(w.store)
+	defer s.Close()
+	ranges, err := util.LoadDeleteRanges(ctx, s, safePoint)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	v2, err := util.IsRaftKv2(ctx, se)
+	v2, err := util.IsRaftKv2(ctx, s)
 	if err != nil {
 		return errors.Trace(err)
 	}
 	// Cache table ids on which placement rules have been GC-ed, to avoid redundantly GC the same table id multiple times.
-	gcPlacementRuleCache := make(map[int64]any, len(ranges))
+	var gcPlacementRuleCache sync.Map
 
 	logutil.Logger(ctx).Info("start delete ranges", zap.String("category", "gc worker"),
 		zap.String("uuid", w.uuid),
 		zap.Int("ranges", len(ranges)))
 	startTime := time.Now()
+
+	if concurrency < 1 {
+		concurrency = 1
+	}
+	concurrencyLimiter := make(chan struct{}, concurrency)
+
+	var wg util2.WaitGroupWrapper
 	for _, r := range ranges {
-		startKey, endKey := r.Range()
-		if v2 {
-			// In raftstore-v2, we use delete range instead to avoid deletion omission
-			task := rangetask.NewDeleteRangeTask(w.tikvStore, startKey, endKey, concurrency)
-			err = task.Execute(ctx)
-		} else {
-			err = w.doUnsafeDestroyRangeRequest(ctx, startKey, endKey, concurrency)
-		}
-		failpoint.Inject("ignoreDeleteRangeFailed", func() {
-			err = nil
-		})
+		concurrencyLimiter <- struct{}{}
+		f := func(r util.DelRangeTask) {
```

Comment:
Need to make sure all the functions called inside the `f` are safe to be executed concurrently, is there any other concurrency risks?

Reply:
I have not identified any additional risks. I have obtained a review for the `doGCLabelRules` from @rleungx.